module/input/Camera/src/Camera.cpp

```diff
@@ -1,129 +1,364 @@
+                            context->time = sync_clocks(arv_camera_get_device(context->camera.get()));
+                            arv_stream_set_emit_signals(context->stream.get(), true);
+                            arv_camera_start_acquisition(context->camera.get());
+                        }
+                    }
                 }
 
-                emit(msg);
-            });
+                // Apply our timestamp offset
+                ts = ts + context->time.offset;
 
-        on<Shutdown>().then([this] {
-            ShutdownV4L2Camera();
-            ShutdownAravisCamera();
-        });
-    }
+                auto msg        = std::make_unique<Image>();
+                msg->format     = context->fourcc;
+                msg->dimensions = Eigen::Matrix<unsigned int, 2, 1>(width, height);
+                // TODO(trent) use an std::vector here to avoid the copy
+                msg->data.insert(msg->data.end(), buff, buff + buffSize);
+                msg->camera_id = context->camera_id;
+                msg->name      = context->name;
+                msg->timestamp = NUClear::clock::time_point(nanoseconds(ts));
+                // TODO attach Hcw once we have access to sensors
+                msg->lens = context->lens;
+
+                context->reactor.emit(msg);
+
+                // Put the buffer back on the queue
+                arv_stream_push_buffer(stream, buffer);
+            }
+            else {
+                std::string msg;
+                switch (arv_buffer_get_status(buffer)) {
+                    case ARV_BUFFER_STATUS_UNKNOWN: msg = "UNKNOWN"; break;
+                    case ARV_BUFFER_STATUS_SUCCESS: msg = "SUCCESS"; break;
+                    case ARV_BUFFER_STATUS_CLEARED: msg = "CLEARED"; break;
+                    case ARV_BUFFER_STATUS_TIMEOUT: msg = "TIMEOUT"; break;
+                    case ARV_BUFFER_STATUS_MISSING_PACKETS: msg = "MISSING PACKETS"; break;
+                    case ARV_BUFFER_STATUS_WRONG_PACKET_ID: msg = "WRONG PACKET ID"; break;
+                    case ARV_BUFFER_STATUS_SIZE_MISMATCH: msg = "SIZE MISMATCH"; break;
+                    case ARV_BUFFER_STATUS_FILLING: msg = "FILLING"; break;
+                    case ARV_BUFFER_STATUS_ABORTED: msg = "ABORTED"; break;
+                }
 
-    // When we shutdown, we must tell our camera class to close (stop streaming)
-    void Camera::ShutdownV4L2Camera() {
-        for (auto& camera : V4L2Cameras) {
-            camera.second.closeCamera();
-            camera.second.disableHandles();
+                // Put the buffer back on the queue and show an error message
+                arv_stream_push_buffer(stream, buffer);
+                NUClear::log<NUClear::WARN>("Failed getting buffer from", context->name, "with error", msg);
+            }
         }
+    }
 
-        V4L2Cameras.clear();
+    void Camera::control_lost(ArvGvDevice*) {
+        NUClear::log<NUClear::WARN>("Control of a camera has been lost ");
```

Comment:
Maybe upgrade this to `NUCLEAR::FATAL` and also get camera details and remove the offending camera from our list (and print out details of which camera is lost)

Reply:
Upgraded to fatal, remember to downgrade this to error one day when we can recover from it